docs/src/examples/BayesianNODE_NUTS.md

```diff
@@ -0,0 +1,254 @@
+
+function trueODEfunc(du, u, p, t)
+    true_A = [-0.1 2.0; -2.0 -0.1]
+    du .= ((u.^3)'true_A)'
+end
+
+prob_trueode = ODEProblem(trueODEfunc, u0, tspan)
+ode_data = Array(solve(prob_trueode, Tsit5(), saveat = tsteps))
+
+
+dudt2 = FastChain((x, p) -> x.^3,
+                  FastDense(2, 50, tanh),
+                  FastDense(50, 2))
+prob_neuralode = NeuralODE(dudt2, tspan, Tsit5(), saveat = tsteps)
+
+function predict_neuralode(p)
+    Array(prob_neuralode(u0, p))
+end
+
+function loss_neuralode(p)
+    pred = predict_neuralode(p)
+    loss = sum(abs2, ode_data .- pred)
+    return loss, pred
+end
+
+l(θ) = -sum(abs2, ode_data .- predict_neuralode(θ)) - sum(θ .* θ)
+
+
+function dldθ(θ)
+    x,lambda = Flux.Zygote.pullback(l,θ)
+    grad = first(lambda(1))
+    return x, grad
+end
+
+metric  = DiagEuclideanMetric(length(prob_neuralode.p))
+
+h = Hamiltonian(metric, l, dldθ)
+
+
+integrator = Leapfrog(find_good_stepsize(h, Float64.(prob_neuralode.p)))
+
+
+prop = AdvancedHMC.NUTS{MultinomialTS, GeneralisedNoUTurn}(integrator)
+
+adaptor = StanHMCAdaptor(MassMatrixAdaptor(metric), StepSizeAdaptor(0.45, prop.integrator))
+
+samples, stats = sample(h, prop, Float64.(prob_neuralode.p), 500, adaptor, 500; progress=true)
+
+
+losses = map(x-> x[1],[loss_neuralode(samples[i]) for i in 1:length(samples)])
+
+##################### PLOTS: LOSSES ###############
+scatter(losses, ylabel = "Loss",  yscale= :log, label = "Architecture1: 500 warmup, 500 sample")
+
+################### RETRODICTED PLOTS: TIME SERIES #################
+pl = scatter(tsteps, ode_data[1,:], color = :red, label = "Data: Var1", xlabel = "t", title = "Spiral Neural ODE")
+scatter!(tsteps, ode_data[2,:], color = :blue, label = "Data: Var2")
+
+for k in 1:300
+    resol = predict_neuralode(samples[100:end][rand(1:400)])
+    plot!(tsteps,resol[1,:], alpha=0.04, color = :red, label = "")
+    plot!(tsteps,resol[2,:], alpha=0.04, color = :blue, label = "")
+end
+
+idx = findmin(losses)[2]
+prediction = predict_neuralode(samples[idx])
+
+plot!(tsteps,prediction[1,:], color = :black, w = 2, label = "")
+plot!(tsteps,prediction[2,:], color = :black, w = 2, label = "Best fit prediction", ylims = (-2.5, 3.5))
+
+#################### RETRODICTED PLOTS - CONTOUR ####################
+pl = scatter(ode_data[1,:], ode_data[2,:], color = :red, label = "Data",  xlabel = "Var1", ylabel = "Var2", title = "Spiral Neural ODE")
+
+for k in 1:300
+    resol = predict_neuralode(samples[100:end][rand(1:400)])
+    plot!(resol[1,:],resol[2,:], alpha=0.04, color = :red, label = "")
+end
+
+plot!(prediction[1,:], prediction[2,:], color = :black, w = 2, label = "Best fit prediction", ylims = (-2.5, 3))
+
+######################## CHAIN DIAGNOSIS PLOTS#########################
+samples = hcat(samples...)
+
+samples_reduced = samples[1:5, :]
+
+samples_reshape = reshape(samples_reduced, (500, 5, 1))
+
+Chain_Spiral = Chains(samples_reshape)
+
+plot(Chain_Spiral)
+
+autocorplot(Chain_Spiral)
+```
```

Comment:
Add the images of the plots here, take a look at the other md files with images for how to do it if you are unfamiliar

Reply:
Yup. Added plots of chain diagnosis as well as the retrodicted plots.